code/script.js

```diff
@@ -229,97 +254,202 @@ const setSecret = () => {
-
+  const { category, questionValue } = currentQuestion
+  let comparison;
+  // new local object with varaibles, gets filled with object currenQuestion from above
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
+  // check values in object with key/value pairs
 
+  if (category === 'hair') {
+    if (questionValue === secret[category]) {
+      // secret is a DOM querey, then placeholder for cat=key
+      comparison = true;
+    } else {
+      comparison = false;
+    }
+    console.log("comparison: " + comparison);
+  } else if (category === "eyes") {
+    if (questionValue === secret[category]) {
+      comparison = true;
+    } else {
+      comparison = false;
+    }
+    console.log("comparison: " + comparison);
+  } else if (category === 'accessories') {
+    if (secret[category].includes(questionValue)) {
+      comparison = true;
+    } else {
+      comparison = false;
+    }
+    console.log("comparison: " + comparison);
+  } else if (category === "other") {
+    if (questionValue === secret.other[0]) {
+      comparison = true;
+    } else {
+      comparison = false;
+    }
+    console.log("comparison: " + comparison);
+    filterCharacters(comparison);
   }
+
+  // This given routine is so overcomplicated, that I discarded it. I took a nice plain "if/else if" above and this is much easier to read and understand. Basta. =]
+  //   if (category === "hair" || category === "eyes"){
+  //   if (questionValue === secret[category]) {
+  //     comparison = true;
+  //   } else {
+  //     comparison = false;
+  //   }
+  //   // check nested arrays in object
+  // } else if (category === 'accessories' || category === 'other') {
+  //   if (category === "other") {
+  //     if (questionValue === secret.other[0]) { comparison = true; }
+  //   } else {
+  //     comparison = false;
+  //   }
+  //   else if (questionValue === "accessories") {
+  //   if (secret[category].includes(questionValue)) {
+  //     comparison = true;
+  //   } else {
+  //     comparison = false;
+  //   }
+  //   console.log(comparison);
+  // }
+  filterCharacters(comparison);
 }
 
-// It'll filter the characters array and redraw the game board.
+// filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  // gets true, false from checkQuestion above
+  counterGuesses += 1;
+  console.log("tries: ", counterGuesses);
+  counterfield.innerHTML = counterGuesses;
+  // update counter field in aside
+  const { category, questionValue } = currentQuestion
```

Comment:
I can see your code doesnt filter away the wrong guesses. perhaps you intended for it to be like this? 

now this might not be correct, but I'm gonna give it a go: 

if you wanted to filter away the wrong guesses: 

//keep would be:
if (category === ''){
if (keep) {charactersInPlay = charactersInPlay.filter((items) => items[category] === questionValue);
......} else {
//filter away:  {charactersInPlay = charactersInPlay.filter((items) => items[category] !== questionValue);
}

I struggled with this quite a bit. But the way i understand it is:  
the ===  filters and keeps only the objects (CHARACTERS) where the value in the category matches the value in the questionvalue and the !== filters away the objects (CHARACTERS)  where the value does not match the questionvalue.